x-pack/plugins/secops/public/components/timeline/timeline.test.tsx

```diff
@@ -271,5 +371,94 @@ describe('Timeline', () => {
         });
       });
     });
+
+    describe('onChangeItemsPerPage', () => {
+      test('it invokes the onChangeItemsPerPage callback when the input is updated', () => {
+        const onChangeItemsPerPage = jest.fn();
+
+        const wrapper = mount(
+          <DragDropContext onDragEnd={noop}>
+            <MockedProvider mocks={mocks}>
+              <Timeline
+                id="foo"
+                activePage={0}
+                columnHeaders={headers}
+                columnRenderers={columnRenderers}
+                dataProviders={mockDataProviders}
+                itemsPerPage={5}
+                itemsPerPageOptions={[5, 10, 20]}
+                onChangeItemsPerPage={onChangeItemsPerPage}
+                onChangePage={noop}
+                onColumnSorted={noop}
+                onDataProviderRemoved={noop}
+                onFilterChange={noop}
+                onRangeSelected={noop}
+                onToggleDataProviderEnabled={noop}
+                pageCount={0}
+                range={'1 Day'}
+                rowRenderers={rowRenderers}
+                sort={sort}
+                width={1000}
+              />
+            </MockedProvider>
+          </DragDropContext>
+        );
+
+        wrapper
+          .find('[data-test-subj="table-pagination"] button')
+          .at(0)
+          .simulate('click');
+
+        wrapper.update();
+
+        wrapper
+          .find('.euiContextMenuItem')
+          .at(0)
+          .simulate('click');
+
+        expect(onChangeItemsPerPage.mock.calls[0][0]).toEqual(5);
+      });
+    });
+
+    describe('onChangePage', () => {
+      // TOOD: This is skipped because right now you cannot get to the rendered paged node
+      // I do not know why but when I open up it up using console.log(wrapper.debug())
+      // I see this non-rendered code instead of it all expanded
+      // <EuiPagination pageCount={0} activePage={0} onPageClick={[Function]}>
+      //   <span />
+      // </EuiPagination>
+      test.skip('it invokes the onChangePage callback when the input is updated', () => {
```

Comment:
Note: I could not complete this one test due to Enzyme not rendering `EuiPagination` children.

Reply:
haha, The test was correct, we found out that EuiTablePagination requires at least 2 pages and updated the code accordingly.